### PR TITLE
Changing the gherkin tests to run on configured launches

### PR DESCRIPTION
**Pull request by jon-betts**
* This caused more errors to be discovered before
 * This required some debugging information in the output
 * Because there's no difference between student and teacher without grading

This is for: https://github.com/hypothesis/lms/issues/1220

### CODE REVIEW

File: lms/views/basic_lti_launch.py [hunk 1]
```diff
@@ -37,8 +37,20 @@ def __init__(self, context, request):
...
+        self.context.js_config.update(
+            {
+                # Configure the front-end mini-app to run.
+                "mode": "basic-lti-launch",
+                # Add debug information (currently used in the gherkin tests)
+                "debug": {
+                    "tags": [
+                        "role:instructor"
+                        if request.lti_user.is_instructor
+                        else "role:learner"
```
Comment (jon-betts) [inline lms/views/basic_lti_launch.py]: Without this there is no way to determine the different between an instructor and learner unless grading is enabled (which it isn't in the LTI tests we've implemented).

File: tests/bdd/steps/h_api.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
+import re
+
+from h_matchers import Any
+from httpretty import httpretty
+
+from tests.bdd.step_context import StepContext
+
+
+class HAPIContext(StepContext):
```
Comment (jon-betts) [inline tests/bdd/steps/h_api.py]: We now try to send requests to H as we have configured launches. We need to accept these gracefully or the tests fail.

File: tests/bdd/steps/feature_steps/lms.feature [hunk 3]
```diff
@@ -4,7 +4,14 @@ Feature: Steps for our app
     Then the response status code is 200
 
   Scenario: the user has instructor privileges
-    Then the response body does not match 'An instructor needs to launch the assignment to configure it.'
+    Then the response body matches 'role:instructor'
```
Comment (lyzadanger) [inline tests/bdd/steps/feature_steps/lms.feature]: This feels a little magical without more context/explanation. Would it be possible to create a new step definition that is more clear that we're looking at returned _debug_ information? Otherwise it's a little inscrutable why `role:instructor` would be in the response body.

File: tests/bdd/steps/__init__.py [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# As this class currently defines no steps it isn't automatically loaded
```
Comment (lyzadanger) [inline tests/bdd/steps/__init__.py]: It took me two passes to grok that _this class_ refers to `HAPIContext`, but maybe I'm just slow on the uptake.

File: tests/bdd/steps/h_api.py (hunk 2 above)
Comment (lyzadanger) [reply tests/bdd/steps/h_api.py]: Aha, yes! Quick crash course for me in `httpretty`. I think I object to the _naming_ of that library as it sounds like a linting tool, not a mocking tool, grin. I wonder if it's worth commenting the `enable` line for nitwits, or whether 99.5% of anyone versed in python should be familiar with it...

File: tests/bdd/steps/feature_steps/lms.feature (hunk 3 above)
Comment (jon-betts) [reply tests/bdd/steps/feature_steps/lms.feature]: I can't think of a robust way of making the step explain where it's getting this from and have it work. I can sure put in a comment though.

File: tests/bdd/steps/__init__.py (hunk 4 above)
Comment (jon-betts) [reply tests/bdd/steps/__init__.py]: I'll reword this to be less confusing.

**Comment by jon-betts** [comment]
I've just rebased this onto the master which has the test running as part of the build. It's showing the error with missing dependencies. Doh! I'll get that fixed.

File: tests/bdd/steps/h_api.py (hunk 2 above)
Comment (jon-betts) [reply tests/bdd/steps/h_api.py]: I've added some comments to the file to say what this is all about.
